src/nbl/video/IDescriptorPool.cpp

```diff
@@ -4,60 +4,63 @@
 namespace nbl::video
 {
 
-core::smart_refctd_ptr<IGPUDescriptorSet> IDescriptorPool::createDescriptorSet(core::smart_refctd_ptr<const IGPUDescriptorSetLayout>&& layout)
+bool IDescriptorPool::createDescriptorSets(uint32_t count, const IGPUDescriptorSetLayout* const* layouts, core::smart_refctd_ptr<IGPUDescriptorSet>* output)
 {
-    if (!isCompatibleDevicewise(layout.get()))
-        return nullptr;
+    core::vector<SDescriptorOffsets> descriptorOffsets(count);
 
-    return createDescriptorSet_impl(std::move(layout));
+    for (uint32_t i = 0u; i < count; ++i)
+    {
+        if (!isCompatibleDevicewise(layouts[i]))
+            return false;
+        
+        descriptorOffsets[i] = allocateDescriptorOffsets(layouts[i]);
```

Comment:
there's one more reason to have that counter of currently live descriptor sets, to validate you're not creating too many (as given by `SCreateInfo::maxSets`)

Reply:
iteratable pool allocator will solve this without a counter (counter is implicit)

Reply:
Resolved.